File: Features/referencestates-sanctionsbody/contracts/src/test/java/net/corda/samples/referencestates/states/SanctionedEntitiesTests.java
package net.corda.samples.referencestates.states;

import net.corda.core.contracts.ContractState;
import net.corda.core.contracts.LinearState;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.wildfly.common.Assert.assertFalse;
import static org.wildfly.common.Assert.assertTrue;

public class SanctionedEntitiesTests {


    TestIdentity a = new TestIdentity(new CordaX500Name("Alice", "", "GB"));
    TestIdentity b = new TestIdentity(new CordaX500Name("Bob", "", "GB"));
    TestIdentity c = new TestIdentity(new CordaX500Name("Charlie", "", "GB"));

    @Test
    public void constructorTest() {

        List<Party> badPeople = Arrays.asList(a.getParty());
        Party issuer = b.getParty();


        SanctionedEntities st = new SanctionedEntities(badPeople, b.getParty());

        assertTrue(st instanceof ContractState);
        assertTrue(st instanceof LinearState);

        assertEquals(badPeople, st.getBadPeople());
        assertEquals(issuer, st.getIssuer());

        assertTrue(st.getParticipants().contains(b.getParty()));
        assertFalse(st.getParticipants().contains(a.getParty()));
        assertFalse(st.getParticipants().contains(c.getParty()));
    }

}

File: kotlin-samples/Features/referenceStates-sanctionsBody/contracts/src/test/kotlin/net/corda/samples/referencestates/states/SanctionedEntitiesTests.kt
package net.corda.samples.referencestates.states

import net.corda.core.contracts.ContractState
import net.corda.core.contracts.LinearState
import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import org.junit.Assert
import org.junit.Test
import java.util.*

class SanctionedEntitiesTests {

    var a = TestIdentity(CordaX500Name("Alice", "", "GB"))
    var b = TestIdentity(CordaX500Name("Bob", "", "GB"))
    var c = TestIdentity(CordaX500Name("Charlie", "", "GB"))

    @Test
    fun constructorTest() {
        val badPeople = Arrays.asList(a.party)
        val issuer = b.party
        val st = SanctionedEntities(badPeople, b.party)
        Assert.assertTrue(st is ContractState)
        Assert.assertTrue(st is LinearState)
        Assert.assertEquals(badPeople, st.badPeople)
        Assert.assertEquals(issuer, st.issuer)
        Assert.assertTrue(st.participants.contains(b.party))
        Assert.assertFalse(st.participants.contains(a.party))
        Assert.assertFalse(st.participants.contains(c.party))
    }
}
